deployment/ccip/changeset/solana/cs_ping_pong.go

```diff
@@ -0,0 +1,254 @@
+}
+
+func (c StartPingPongConfig) Validate(e deployment.Environment) error {
+	return nil
+}
+func StartPingPongContractChangeset(
+	e deployment.Environment,
+	c StartPingPongConfig,
+) (deployment.ChangesetOutput, error) {
+	if err := c.Validate(e); err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("invalid StartPingPongConfig: %w", err)
+	}
+
+	s, err := ccipChangeset.LoadOnchainStateSolana(e)
+	if err != nil {
+		e.Logger.Errorw("Failed to load existing onchain state", "err", err)
+		return deployment.ChangesetOutput{}, err
+	}
+
+	chain := e.SolChains[c.ChainSelector]
+	chainState := s.SolChains[c.ChainSelector]
+
+	ppConfigPDA, _, err := solanaStateUtils.FindPingPongDemoConfigPDA(chainState.PingPong)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	ppSendSignerPDA, _, err := solanaStateUtils.FindPingPongCCIPSendSignerPDA(chainState.PingPong)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	destChainStatePDA, err := solanaStateUtils.FindDestChainStatePDA(c.ChainSelector, chainState.Router)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	routerNoncePDA, err := solanaStateUtils.FindNoncePDA(c.ChainSelector, solana.PublicKey(*chain.DeployerKey), chainState.Router)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	feeBillingSignerPDA, _, err := solanaStateUtils.FindFeeBillingSignerPDA(chainState.Router)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	fqBillingTokenConfigPDA, _, err := solanaStateUtils.FindFqBillingTokenConfigPDA(nil, chainState.FeeQuoter)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	rmnRemoteConfigPDA, _, err := solanaStateUtils.FindRMNRemoteConfigPDA(chainState.RMNRemote)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	rmnRemoteCursesPDA, _, err := solanaStateUtils.FindRMNRemoteCursesPDA(chainState.RMNRemote)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	initTx, err := solPingPong.NewStartPingPongInstruction(
+		ppConfigPDA,
+		chain.DeployerKey.PublicKey(),
+		ppSendSignerPDA,
+		nil, // feeTokenProgram ag_solanago.PublicKey,
+		nil, // feeTokenMint ag_solanago.PublicKey,
+		nil, // feeTokenAta ag_solanago.PublicKey,
+		chainState.Router,
+		chainState.RouterConfigPDA,
+		destChainStatePDA,
+		routerNoncePDA,
+		chainState.Receiver, // TODO is this right?
+		feeBillingSignerPDA,
+		chainState.FeeQuoter,
+		chainState.FeeQuoterConfigPDA,
+		nil, // feeQuoterDestChain ag_solanago.PublicKey,
```

Comment:
This should be the PDA belonging to fee quoter that holds the configuration for the destination chain. See [the rust code](https://github.com/smartcontractkit/chainlink-ccip/blob/main/chains/solana/contracts/programs/ccip-router/src/context.rs#L370-L375) defining what is required there and the [go code](https://github.com/smartcontractkit/chainlink-ccip/blob/d2fd3633f4854492d233a0b545dd17895cfcdc4e/chains/solana/contracts/tests/config/ccip_config.go#L77) calculating it for the integration test (keep in mind you're probably using a different chain selector for the dest chain here)